### ISSUE TITLE
Generating RAML at build time

### ISSUE THREAD

**Issue by davidkeaveny** [mulesoft-labs/raml-dotnet-tools#73]
Are there any plans to allow the generation of RAML at build time? I love RAML for contract-first development, but right now I am working on an existing API for which I need to take a code-first approach.

Now while the RamlApiExplorer code is great for exposing the RAML definition of your API when the API is running, I want to be able to generate RAML as part of my build process, so that definition can be used to e.g. generate clients in C# and TypeScript as part of that build, so that I can start consuming the latest API feature in my apps without having to wait for the API to be hosted somewhere. I can also then save those definitions and diff with previous versions to identify breaking changes in backwards compatibility, which could even be used to fail the build (how awesome is that?).

So in other words, I would really like there to be a .exe (or even a .dll with some MSBuild .targets) that I can call in the AfterBuild phase of an MSBuild .csproj, point it at one or more DLLs that contains my API controllers and DTOs, and have a single RAML definition generated from that.

Does that scratch anyone else's itch?

**Comment by woodp** [comment]
Hi David, we have built the [command line tool](https://github.com/mulesoft-labs/raml-dotnet-tools/tree/master/command-line) exactly for that.
Let us know how it goes for you.

**Comment by davidkeaveny** [comment]
The command line tool is great _if_ you already have a RAML definition and want to create a client or controller; but as I said, I'm working in a code-first environment where I already have my controllers, and I want to  be able to create the RAML definition _at build-time_, so that other steps in my build process can consume that definition (no doubt using the command line tool that you mention).

**Comment by woodp** [comment]
will consider your request David, thanks for the feedback
